src/main/java/org/zalando/putittorest/HttpAsyncClientFactoryBean.java

```diff
@@ -24,10 +29,19 @@ public void setLastResponseInterceptors(final List<HttpResponseInterceptor> inte
         interceptors.forEach(builder::addInterceptorLast);
     }
 
+    @Hack("Hide type to avoid runtime dependency")
+    public void setMetricsWrapper(final Object metricsWrapper) {
+        this.metricsWrapper = metricsWrapper;
+    }
+
     @Override
     public HttpAsyncClient getObject() {
         // TODO: builder.setConnectionTimeToLive(30, TimeUnit.SECONDS);
-        return builder.build();
+        final CloseableHttpAsyncClient client = builder.build();
+        if (metricsWrapper != null) {
+            return new InstrumentedHttpAsyncClient((MetricsWrapper) metricsWrapper, client);
```

Comment:
Yes he will not try to load the class (at least that's what he did when I tested it). If you type the field with `MetricsWrapper` the class will be loaded just because Spring things it is a good idea to check out all the fields of your `FactoryBean`.